### PR TITLE
[Merged by Bors] - Fix cPool flaky unit test

### PR DESCRIPTION
**Pull request by y0sher**
## Motivation
Unit test was failing randomly .

## Changes

better check the errors and wait for connections to close before ending the test.

### CODE REVIEW

**Review by noamnelke — APPROVED** [review]
I'm not a big fan of randomness in unit tests, but if you believe that removing the seeding was "necessary evil" we could add it back, but print the seed. This way, if it randomly fails we can hardcode the random seed into the test to reproduce the failure.

Approved with/without the change I proposed. Up to you.

**Comment by y0sher** [comment]
> I'm not a big fan of randomness in unit tests, but if you believe that removing the seeding was "necessary evil" we could add it back, but print the seed. This way, if it randomly fails we can hardcode the random seed into the test to reproduce the failure.
> 
> Approved with/without the change I proposed. Up to you.

I think you missed something or misread the PRs, in this PR I haven't changed anything related to randomness.

**Comment by y0sher** [comment]
bors merge

File: p2p/connectionpool/connectionpool_test.go [hunk 1]
```diff
@@ -268,7 +269,6 @@ func TestRandom(t *testing.T) {
 	nMock.SetDialDelayMs(50)
 	nMock.SetDialResult(nil)
 	cPool := NewConnectionPool(nMock.Dial, generatePublicKey(), log.NewDefault(t.Name()))
-	rand.Seed(time.Now().UnixNano())
```
Comment (noamnelke) [inline p2p/connectionpool/connectionpool_test.go:271]: by not seeding you removed the randomness, no?

**Comment by y0sher** [comment]
bors merge

**Pull request closed without merging**